worker/worker.go

```diff
@@ -196,6 +203,42 @@ func (w *Worker) EmitEvent(
 		args...)
 }
 
+// cancel implements com.redhat.Yggdrasil1.Worker1.Cancel method by calling the
+// worker's cancelRxFunc in a goroutine.
+func (w *Worker) cancel(addr string, id string, cancelID string) *dbus.Error {
+	// If worker doesn't implements the cancelation function it does nothing
+	if w.cancelRx == nil {
+		log.Debug("worker does not support cancellation messages")
+		return dbus.NewError(
+			"org.freedesktop.DBus.UnkownInterface",
+			[]interface{}{"cancel method not implemented"},
+		)
+	}
+
+	log.Tracef("cancelling message:")
+	log.Tracef("addr = %v", addr)
+	log.Tracef("id = %v", id)
+	log.Tracef("cancel-id = %v", cancelID)
+
+	// Communicate the worker accepts the message and its starting to
+	// work on it.
+	if err := w.EmitEvent(ipc.WorkerEventNameBegin, id, "", map[string]string{}); err != nil {
+		return dbus.NewError("com.redhat.Yggdrasil1.Worker1.EventError", []interface{}{err.Error()})
+	}
+
+	go func() {
+		if err := w.cancelRx(w, addr, id, cancelID); err != nil {
+			log.Errorf("cannot call cancelRx: %v", err)
```

Comment:
I would re-phrase this error message a little, because it is not 100% correct. I would use something like this:

```go
log.Errorf("callback function cancelRx() was terminated: %v", err)
```